pkg/v1/remote/index.go

```diff
@@ -182,3 +182,48 @@ func (r *remoteIndex) childDescriptor(child v1.Descriptor, platform v1.Platform)
 		platform:   platform,
 	}, nil
 }
+
+// Check if the given platform can run the required platform.
+// Matching platforms must have identical architecture, OS,
+// OS version (if provided), and variant (if provided)
+// Required features must be a subset of the features in the given platform.
+func (r *remoteIndex) matchesPlatform(given v1.Platform, required v1.Platform) bool {
```

Comment:
I think this function doesn't have to be a function called on *remoteIndex.

And for args you can do (given, required v1.Platform)